lib/Doctrine/ORM/Persisters/Collection/CollectionPersister.php

```diff
@@ -15,6 +15,8 @@ interface CollectionPersister
     /**
      * Deletes the persistent state represented by the given collection.
      *
+     * @param PersistentCollection<array-key, object> $collection
```

Comment:
Why's this necessary?

Reply:
I think template parameters should always be specified, shouldn't they?

Reply:
Yes, but you're narrowing the second type parameter from `mixed` to `object`.

Reply:
I don't remember, and my commit message doesn't help, sorry.
When I try running Psalm v5 with all the errors baselined and then changing `object` to `mixed`, I get this error:

```
ERROR: InvalidArgument - lib/Doctrine/ORM/UnitOfWork.php:425:94 - Argument 1 of Doctrine\ORM\Persisters\Collection\CollectionPersister::delete expects Doctrine\ORM\PersistentCollection<array-key, mixed>, but Doctrine\ORM\PersistentCollection<array-key, object> provided (see https://psalm.dev/004)
                    $this->getCollectionPersister($collectionToDelete->getMapping())->delete($collectionToDelete);
```

Maybe the right fix would be to use `@template-covariant` here… also, I wonder if we should use `T of object` for as a template for `PersistentCollection`… a `PersistentCollection` is always going to be a collection of entities, isn't it?

Reply:
> Maybe the right fix would be to use `@template-covariant` here…

We can't for the same reasons we've introduced the `ReadableCollection`. 😕 

> also, I wonder if we should use `T of object` for as a template for `PersistentCollection`… a `PersistentCollection` is always going to be a collection of entities, isn't it?

Probably, but then again, do `CollectionPersister` implementations actually care about the type of the values?

Did you try to declare the methods as generic methods instead, like…

```php
/**
 * @template TKey of array-key
 * @template TValue
 *
 * @param PersistentCollection<TKey, TValue> $collection
 *
 * @return void
 */
public function delete(PersistentCollection $collection);
```

Reply:
> We can't for the same reasons we've introduced the ReadableCollection. 

Ah yeah right, I should know :sweat: 

> Probably, but then again, do CollectionPersister implementations actually care about the type of the values?

I suppose they don't, since specifying that we have an object here does not improve the baseline more than silencing the aforementioned error.

> Did you try to declare the methods as generic methods instead, like…

That appears to work, thanks a lot! I don't understand why we have this issue only with `delete()` though. I didn't notice the methods below before noticing the plural to "methods" in your previous message, and now I have mixed feeling about this… probably just going to drop that commit as well, so that we find a better solution later.